h2/src/main/org/h2/value/ValueTimestampTimeZone.java

```diff
@@ -202,27 +203,16 @@ public short getTimeZoneOffsetMins() {
     }
 
     /**
-     * Returns name of the timezone such as UTC or GMT+1:00.
+     * Returns compatible offset-based time zone with no DST schedule.
```

Comment:
since you are working on this, and seem to have a need for it, do you forsee any problems with DST given that (unlike some other DBs) we do not preserve the name of the original timezone for the timestamp?

Reply:
It depends on what exactly somebody want to do with them. For accurate storage current approach is suitable for modern timezones (for historic limiting precision of offset to minutes is not enough, but it should not be needed for normal usage). For comparison its enough too.

But with functions like `DATEADD` situation are complex. H2 have bugs even with plain `TIMESTAMP` types in some timezones. Even `java.util.Calendar` have own bugs and H2 already have some workarounds for some of them (but sometimes tests fail in my timezone depending on the current date). JSR-310 types on Java 8 are much more reliable, but we can't use them now.

`Types.TIMESTAMP_WITH_TIMEZONE` from JDBC is mapped to `java.time.OffsetDateTime` that also stores only offset. There is no `java.time.ZonedDateTime` equivalent in JDBC, but some databases have such types.

If we introduce `ZonedDateTime` equivalent in H2, we may got weird issues on `tzdb` changes. This type will not be safe for comparison or for usage in indexes.